Wazo 2671 wazo confd cli

pc-m opened the pull request:
This PR should be merged after https://github.com/wazo-platform/xivo-tools/pull/42

pc-m committed:
boilerplate: add boilerplate for the CLI

pc-m committed:
commands: add "user list"

pc-m committed:
commands: add "endpoint sip list"

pc-m committed:
commands: add "endpoint sip template list"

pc-m committed:
endpoint sip list: add "--template <uuid>" to filter by template

pc-m committed:
add a tox.ini file

pc-m committed:
add a zuul config

pc-m committed:
add Debian packaging

pc-m commented:
recheck

pc-m commented:
recheck

pc-m committed:
add an example to the readme

sduthil commented on wazo_confd_cli/commands/endpoint.py:
```diff
@@ -0,0 +1,114 @@
+        raw_items = result['items']
+        if parsed_args.template:
+
+            def predicate(item):
```
```suggestion
            def has_template(item):
```

sduthil committed:
add dockerfile

sduthil commented on wazo_confd_cli/commands/endpoint.py:
```diff
@@ -0,0 +1,114 @@
+                        return True
+                return False
+
+            raw_items = [item for item in raw_items if predicate(item)]
```
```suggestion
            raw_items = [item for item in raw_items if has_template(item)]
```

sduthil reviewed (changes requested):
I think a `--prefix` option would be nice too, in order to use `wazo-confd-cli` from another machine, e.g. with Docker:
```
docker run -it wazoplatform/wazo-confd-cli --hostname wazo.example.com --port 443 --prefix /api/confd --token mytoken user list
```

sduthil commented:
Also, when trying the `wazo-confd-cli`, there is always an error message at the start (even though the CLI commands are working):

```
Could not read config dir /root/.config/wazo-confd-cli: [Errno 2] No such file or directory: '/root/.config/wazo-confd-cli'
```

sduthil commented on wazo_confd_cli/commands/endpoint.py:
```diff
@@ -0,0 +1,114 @@
+                return False
+
+            raw_items = [item for item in raw_items if predicate(item)]
+        headers = self.extract_column_headers(raw_items[0])
```
There is no validation that `raw_items` is empty.
If I run `wazo-confd-cli ... endpoint sip list --template does-not-exist`, then I get a Traceback on this line.

pc-m committed:
lists: on empty result return the known columns

Adding the columns avoid the error when using `-c` flag and no results are
returned.

pc-m committed:
config: check if the user config exists before reading it

the packaging does not create a user directory for this configuration. this
generates an error message when using the CLI even if everything works

pc-m committed:
args: add --prefix

this allows the CLI to be used from another host and use the /api/confd API
prefix

pc-m committed:
arg parser: split https from certificate options

this change allow the user to specify --ssl AND --verify before this change all
options were mutually exclusive making it impossible to change the default https
and verify_certificate argument when launching the command

pc-m commented:
Thanks for the review, I fixed all reported problems. Some of them have been force pushed into the commit that adds the `--template` argument. I also changed the parsing of the https options because we could not specify the `https` and `verify_certificate` options that were in a mutually exclusive group
